### DISCUSSION TITLE
Best of both

### DISCUSSION THREAD

**Discussion by teledactyl** [microsoft/teams-sdk#1616] [Q&A]
I've developed a nice Bot using 'sequence' augmentation that can automate several things for me in my stage-view app within Teams.
I also want it to act as an experienced engineer so it can begin to give advice, so I've used Vectra to encode a few documents on a fairly narrow engineering discipline. When using 'none' as augmentation, I get some good, detailed, results.
But, when switching back to 'sequence', the advice becomes quite terse.
![image](https://github.com/microsoft/teams-ai/assets/130683546/30e8c06d-7dc1-4fff-86bb-b34cf89fa95c)
In VS code's terminal I can see that the prompt is picking up similar content, but the results are consistently different.
How can we get the best of both worlds?

**Comment by lilyydu** [comment]
As per live discussion, additional samples to consider for encoding additional docs/information

https://github.com/microsoft/teams-ai/tree/main/js/samples/04.ai-apps/h.datasource-azureOpenAI
https://github.com/microsoft/teams-ai/tree/main/js/samples/04.ai-apps/g.datasource-azureAISearch

**Reply by Carter425** [reply]
Here is the form to apply to Azure OpenAI service 

https://customervoice.microsoft.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR7en2Ais5pxKtso_Pz4b1_xUNTZBNzRKNlVQSFhZMU9aV09EVzYxWFdORCQlQCN0PWcu

**Reply by Carter425** [reply]
Here is the sample for Azure OpenAI On Your Data https://github.com/microsoft/teams-ai/tree/main/js/samples/04.ai-apps/h.datasource-azureOpenAI

**Reply by Carter425** [reply]
Here is a nice video walkthrough to go along with the tutorial: https://www.youtube.com/watch?v=VLP32Ci02M0

**Comment by singhk97** [comment]
Hi @teledactyl, you can also try adding a few examples of user/model interaction within the prompt. This technique is called few-shot prompting, and it's really effective in guiding the model to return responses in a specific format. 

* https://www.promptingguide.ai/techniques/fewshot

**Reply by teledactyl** [reply]
I love this prompting guide: it's a real eye-opener.
